doc/api/child_process.markdown

```diff
@@ -392,76 +392,202 @@ controlling terminal.
-   file descriptor. Setting this option enables the ChildProcess.send() method.
-   If the child writes JSON messages to this file descriptor, the
-   `ChildProcess.on('message')` event handler will be triggered in the parent.
-   If the child is a Node.js process, the presence of an IPC channel will enable
-   `process.send()`, `process.disconnect()`, `process.on('disconnect')`, and
-   `process.on('message')` within the child.
-3. `'ignore'` - Instructs Node.js to ignore the fd in the child. While Node.js
-   will always open fds 0 - 2 for the processes it spawns, setting the fd to
-   `'ignore'` will cause Node.js to open `/dev/null` and attach it to the
-   child's fd.
-4. `Stream` object - Share a readable or writable stream that refers to a tty,
-   file, socket, or a pipe with the child process. The stream's underlying
-   file descriptor is duplicated in the child process to the fd that
-   corresponds to the index in the `stdio` array. Note that the stream must
-   have an underlying descriptor (file streams do not until the `'open'`
-   event has occurred).
-5. Positive integer - The integer value is interpreted as a file descriptor
-   that is is currently open in the parent process. It is shared with the child
-   process, similar to how `Stream` objects can be shared.
-6. `null`, `undefined` - Use default value. For stdio fds 0, 1 and 2 (in other
-   words, stdin, stdout, and stderr) a pipe is created. For fd 3 and up, the
-   default is `'ignore'`.
+The `options.stdio` property is used to configure the [standard streams][] of a
+child process. This allows a child process to receive input from devices like
+the keyboard and send output to devices like the terminal. Configuring
+`options.stdio` also allows the parent to connect a child to additional
+endpoints like files and sockets.
 
-Example:
+The `options.stdio` property accepts an `Array` where each index corresponds to
+the child's standard stream file descriptor. Indices 0, 1, and 2 correspond to
+the stdin, stdout, and stderr streams respectively. Indices 3 and up can be used
+to configure non-standard streams.
+
+The following are the possible values for the `options.stdio` configuration
+`Array`.
+
+* `Integer` file descriptor
+* `Stream` object
+* `'ignore'`
+* `'ipc'`
+* `'pipe'`
+* `null`
+* `undefined`
+
+For convenience, the `options.stdio` property also accepts one of the following
+`String` shorthands.
+
+| Shorthand   | Equivalent                       |
+|-------------|----------------------------------|
+| `'ignore'`  | `['ignore', 'ignore', 'ignore']` |
+| `'inherit'` | `[0, 1, 2]`                      |
+| `'pipe'`    | `['pipe', 'pipe', 'pipe']`       |
+
+When no value is provided, `options.stdio` defaults to `'pipe'`.
+
+##### `Integer` file descriptor
+
+Shares the parent process's open `Integer` file descriptor with a child process.
+The connection to the parent's file descriptor resource is duplicated and
```

Comment:
there is no connection or attachment: "The parent's file descriptor is duplicated in the child to a new descriptor with the value of the Array index."